src/array.cr

```diff
@@ -1471,9 +1471,12 @@ class Array(T)
   # Append multiple values. The same as `push`, but takes an arbitrary number
   # of values to push into the array. Returns `self`.
   def push(*values : T)
-    values.each do |value|
-      self << value
+    new_size = @size + values.size
+    resize_to_capacity(Math.pw2ceil(new_size))
```

Comment:
This calls `resize_to_capacity` each time you push, but I think this should be called only when needed (`if new_size > @capacity`). Same goes for unshift.